### PR TITLE
Fix go.opentelemetry.io/contrib CVE

### PR DESCRIPTION
**Pull request by DharmitD**
## The issue resolved by this Pull Request:
Resolves https://issues.redhat.com/browse/RHOAIENG-2259

## Description of your changes:

Fixing go.opentelemetry.io/contrib dependency related snyk CVE.
Made this change on my fork (correctly rebased to and snyced with the remote's main) 
Ran `go mod tidy` 
Imported my fork in synk and made sure the relevant CVEs were fixed with this update.

## Testing instructions

Deploy a pipeline and make sure everything works as expected.

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -129,3 +129,5 @@ require (
 )
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.56.3
+
+replace go.opentelemetry.io/contrib v0.20.0 => go.opentelemetry.io/contrib v0.44.0
```
Comment (HumairAK) [inline go.mod:133]: can you move this in a single block

Comment (DharmitD) [reply go.mod]: Done, moved them in a single block.

**Comment by gregsheremeta** [comment]
> Testing instructions
> Deploy a pipeline and make sure everything works as expected.

Since this is DSPO, deploying a pipeline won't use the changed code. Making sure DSPAs stand up properly will :)

**Comment by gregsheremeta** [comment]
code lgtm

i didn't verify it ... @gmfrasca will

**Review by gmfrasca — COMMENTED** [review]
/lgtm

**Comment by HumairAK** [comment]
some additional info on this one, 
it's mostly the k8s packages utilizing this dependency, at some point we will just want to update the k8s libraries to a newer k8s (supported by earliest supported ocp version) that utilizes this updated dependency so we can remove it from `replace`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-560`

**Pull request merged**